src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/14.diff.sql

```diff
@@ -0,0 +1,542 @@
+/*************************************************************
+    Resource change capture feature
+**************************************************************/
+
+/*************************************************************
+    Resource change data table
+**************************************************************/
+
+IF NOT EXISTS (
+    SELECT 1 
+    FROM sys.tables
+    WHERE name = 'ResourceChangeData')
+BEGIN
+    CREATE TABLE dbo.ResourceChangeData
+    (
+        Id bigint IDENTITY(1,1) NOT NULL,
+        Timestamp datetime2(7) NOT NULL,
+        ResourceId varchar(64) NOT NULL,
+        ResourceTypeId smallint NOT NULL,
+        ResourceVersion int NOT NULL,
+        ResourceChangeTypeId smallint NOT NULL,
+     CONSTRAINT PK_ResourceChangeData PRIMARY KEY CLUSTERED 
+    (
+        [Id] ASC
+    )WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
+    ) ON [PRIMARY]
+END
+GO
+
+IF EXISTS (
+  SELECT 1 FROM information_schema.columns WHERE table_name = 'ResourceChangeData'
+and column_name = 'Timestamp'
+and Table_schema = 'dbo'
+and column_default is NULL)
+BEGIN
+     ALTER TABLE dbo.ResourceChangeData ADD  CONSTRAINT DF_ResourceChangeData_Timestamp  DEFAULT (sysutcdatetime()) FOR Timestamp
+END
+GO
+
+
+/*************************************************************
+    Resource change type table
+**************************************************************/
+
+IF NOT EXISTS (
+    SELECT 1 
+    FROM sys.tables
+    WHERE name = 'ResourceChangeType')
+BEGIN
+
+     CREATE TABLE dbo.ResourceChangeType(
+         ResourceChangeTypeId smallint IDENTITY(0,1) NOT NULL,
+         Name nvarchar(50) NOT NULL,
+      CONSTRAINT PK_ResourceChangeType PRIMARY KEY CLUSTERED 
+     (
+         ResourceChangeTypeId ASC
+     )WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY],
+      CONSTRAINT UQ_ResourceChangeType_Name UNIQUE NONCLUSTERED 
+     (
+         Name ASC
+     )WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON, OPTIMIZE_FOR_SEQUENTIAL_KEY = OFF) ON [PRIMARY]
+     ) ON [PRIMARY]
+
+     SET IDENTITY_INSERT [dbo].[ResourceChangeType] ON 
```

Comment:
What's the reason for making ResourceChangeTypeId an identity column but then inserting ids directly anyway?

Reply:
ResourceChangeTypeId is the identity and primary key for preventing duplications. Wanted to specify explicitly that 0 for creation, 1 for the update, and 2 for deletion.

Reply:
But why does it need to be identity?  Just being a PK will prevent duplicates.  Are we planning to insert new rows in the future without specifying the id?

Reply:
Good point. No, not planning to insert new rows without specifying the id. Removed the IDENTITY property.